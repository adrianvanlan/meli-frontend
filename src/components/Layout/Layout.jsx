import AppBar from '../AppBar/AppBar';
import { Outlet } from 'react-router-dom';
import styles from './Layout.module.scss';

function Layout() {
  return (
    <>
      <AppBar />
      <div className={styles.Layout__container}>
        <Outlet />
      </div>
    </>
  );
}

export default Layout;
