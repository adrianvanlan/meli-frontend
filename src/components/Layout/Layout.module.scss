@import '../../styles/variables';
@import '../../styles/mixins';

.Layout__container {
  position: relative;
  margin: 0 auto;
  padding-right: 1rem;
  padding-left: 1rem;

  @include mq(sm) {
    max-width: $col-sm;
  }

  @include mq(md) {
    max-width: $col-md;
  }

  @include mq(lg) {
    max-width: $col-lg;
  }

  @include mq(xl) {
    max-width: $col-xl;
  }
}
