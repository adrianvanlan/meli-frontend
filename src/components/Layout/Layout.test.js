import Layout from './Layout';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

describe('Should render Layout', () => {
  it('should render Layout', () => {
    const wrapper = render(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
