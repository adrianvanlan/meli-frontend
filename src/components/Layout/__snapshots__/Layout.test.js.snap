// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Layout should render Layout 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="AppBar"
      >
        <div
          class="AppBar__Container"
        >
          <a
            class="Logo__Link"
            href="/"
          >
            <img
              alt="Logo Mercadolibre"
              src="Logo_ML.png"
              srcset="Logo_ML.png 53w, Logo_ML@2x.png.png 106w"
              width="53"
            />
          </a>
          <form
            class="SearchInput"
          >
            <input
              autocomplete="off"
              class="SearchInput__input"
              name="q"
              placeholder="Buscar productos, marcas y más…"
              value=""
            />
            <button
              class="SearchInput__button"
              type="submit"
            >
              <img
                alt="Icono buscar"
                src="ic_Search@2x.png.png"
                srcset="ic_Search.png 18w, ic_Search@2x.png.png 36w"
                width="18"
              />
            </button>
          </form>
        </div>
      </header>
      <div
        class="Layout__container"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="AppBar"
    >
      <div
        class="AppBar__Container"
      >
        <a
          class="Logo__Link"
          href="/"
        >
          <img
            alt="Logo Mercadolibre"
            src="Logo_ML.png"
            srcset="Logo_ML.png 53w, Logo_ML@2x.png.png 106w"
            width="53"
          />
        </a>
        <form
          class="SearchInput"
        >
          <input
            autocomplete="off"
            class="SearchInput__input"
            name="q"
            placeholder="Buscar productos, marcas y más…"
            value=""
          />
          <button
            class="SearchInput__button"
            type="submit"
          >
            <img
              alt="Icono buscar"
              src="ic_Search@2x.png.png"
              srcset="ic_Search.png 18w, ic_Search@2x.png.png 36w"
              width="18"
            />
          </button>
        </form>
      </div>
    </header>
    <div
      class="Layout__container"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
